src/main/java/org/webshar/hrms/repository/Impl/EmployeeRepositoryImpl.java

```diff
@@ -0,0 +1,108 @@
+    CriteriaQuery<Employee> criteriaQuery = employeeCriteriaQuery.select(employeeRoot);
+    TypedQuery<Employee> typedQuery = entityManager.createQuery(criteriaQuery)
+        .setFirstResult(
+            (employeeSearchRequest.getPage() - 1) * employeeSearchRequest.getPerPage())
+        .setMaxResults(employeeSearchRequest.getPerPage());
+    return typedQuery;
+  }
+
+  private void buildCommonQuery(EmployeeSearchRequest employeeSearchRequest, CriteriaBuilder criteriaBuilder,
+      CriteriaQuery<Employee> employeeCriteriaQuery, Root<Employee> employeeRoot) {
+
+    List<Predicate> andPredicates = new ArrayList<>();
+    List<Predicate> orPredicates = new ArrayList<>();
+
+    if (!StringUtils.isBlank(employeeSearchRequest.getQueryString())) {
+      orPredicates.add(
+          criteriaBuilder.like(criteriaBuilder.lower(employeeRoot.get("firstName")),
+              "%" + employeeSearchRequest.getQueryString().toLowerCase() + "%"));
+      orPredicates.add(
+          criteriaBuilder.like(criteriaBuilder.lower(employeeRoot.get("middleName")),
+              "%" + employeeSearchRequest.getQueryString().toLowerCase() + "%"));
+      orPredicates.add(
+          criteriaBuilder.like(criteriaBuilder.lower(employeeRoot.get("lastName")),
+              "%s" + employeeSearchRequest.getQueryString().toLowerCase() + "%"));
+    }
+
+    if (employeeSearchRequest.getActive() != null) {
+      if (employeeSearchRequest.getActive()) {
+        andPredicates.add(criteriaBuilder.isTrue(employeeRoot.get("isActive")));
+      } else {
+        andPredicates.add(criteriaBuilder.isFalse(employeeRoot.get("isActive")));
+      }
+    }
+
+    String sortBy = getSortByValue(employeeSearchRequest);
+
+    switch (Order.valueOf(employeeSearchRequest.getOrder())) {
+      case ASC:
+        employeeCriteriaQuery.orderBy(criteriaBuilder.asc(employeeRoot.get(sortBy)));
+        break;
+      case DESC:
+        employeeCriteriaQuery.orderBy(criteriaBuilder.desc(employeeRoot.get(sortBy)));
+        break;
+    }
+
+    if (orPredicates.size() == 0) {
+      Predicate andPredicate = criteriaBuilder
+          .and(andPredicates.toArray(new Predicate[andPredicates.size()]));
+      employeeCriteriaQuery.where(andPredicate);
+    } else {
+      Predicate andPredicate = criteriaBuilder
+          .and(andPredicates.toArray(new Predicate[andPredicates.size()]));
+      Predicate orPredicate = criteriaBuilder.or(orPredicates.toArray(new Predicate[orPredicates.size()]));
+      employeeCriteriaQuery.where(andPredicate, orPredicate);
+    }
+  }
+
+  public static String getSortByValue(final EmployeeSearchRequest employeeSearchRequest) {
+    String sortBy = "employeeId";
+    switch (EmployeeSortBy.valueOf(employeeSearchRequest.getSortBy())) {
+      case EMPLOYEE_FIRST_NAME:
+        sortBy = "firstName";
+        break;
+      case JOINING_DATE:
+        sortBy = "joiningDate";
+        break;
+    }
+    return sortBy;
+  }
+}
```

Comment:
There are a few code-smells suggested by sonar lint link missing default, incorrect package naming convention, etc.
Please have a look. The rest all looks fine from functionality point of view.

Reply:
Done there were 86 such a suggestions addressed all of them also formatted all java files with new formatter

Reply:
Thanks, That's quite a heavy lifting. 👍